### DISCUSSION TITLE
Allgemeine Frage zum Setup / Blitzreflektionen

### DISCUSSION THREAD

**Discussion by AdmiralRaccoon** [jomjol/AI-on-the-edge-device#1403] [General]
### Issue

Moin,
eine allgemeine Frage zum Setup mit den ESP32CAMS: Egal wie ich es drehe und wende, ich hab immer eine ungünstige Reflektion im Bild, sowohl beim Gaszähler als auch bei der Wasseruhr (Siehe Bilder).
Beim Gaszähler hab ich dann einfach (vorerst) auf die Nachkommastellen verzichtet. Bei der Wasseruhr ist allerdings genau auf der letzten Ziffer eine Reflektion, und kann vermutlich darum im aktuellen Fall die 8 nicht erkennen?

Hattet Ihr damit ähnliche Probleme? Ich hätte jetzt überlegt das mit einer Art Diffusor über der Blitz-LED zu lösen, aber hab noch keine Idee wie genau.

Danke und Gruss,
Philipp

![image](https://user-images.githubusercontent.com/41727850/203731429-4ac5cba2-3e59-4c12-9fb1-4fd39ea9d033.png)
![image](https://user-images.githubusercontent.com/41727850/203731536-dba2856f-30a2-4fb7-9d6b-dddad1798cad.png)

**Comment by Ingo-Knito** [comment]
Ich hatte exakt die gleichen Probleme und habe lange daran herumgedoktert: 'Das muss doch gehen...' ;-)

Final klappte es über externe WS2812x-LEDs. Es reichen dafür ein oder max. zwei LED die man über drei Leitungen an den ESP anschließt: 5V, GND und GPIO12. Ein Widerstand oder Kapazität ist nicht notwendig und funktioniert bei mir seit Monaten zuverlässig an vier ESP-CAMs.

Die LED gibt es als Meterware für wenige € und sind problemlos zu verkleben bzw. montieren. Beim Löten auf die Richtung achten. Es gibt dafür einen Pfeil, sonst leuchten die LED wegen der seriellen Ansteuerung nicht. Bei der Montage einen Platz aussuchen bei dem der Winkel absichtlich ungüstig zur Linse ist und dann keine Reflexionen auftauchen. Am Besten seitlich, etwas erhöht.

Eingerichtet ist das dann extrem simpel in der Konfiguration über die "GPIO Settings". Haken setzen und gut :-)

Damit macht das Ganze echt Spaß und fehlerhafte Erkennungen werden extrem selten.

**Comment by AdmiralRaccoon** [comment]
Ah super, Danke 👍 
Das klingt ganz praktisch, blöderweise bin ich was Löten angeht wegen meine schlechten Augen so gut wie raus - es war für mich schon ein Krampf den Kleber von Kamera zu lösen um den Fokus verstellen zu können. :-(

Mir sind dann noch 3 weitere Ideen eingefallen, sagt mir ehrlich ob davon was für Euch in Frage kommen würde, bzw. ob man aus 1.1 ein Feature Request machen könne:


1.) Für 2,39€ einen weissen USB LED Strip reinkleben der einfach permanent leuchtet?

1.1) DIesen Strip in eine smarte Steckdose packen und über eine Automation vom Home Assistant einschalten, wenn Ai-on-the-edge das Foto macht. Das ist natürlich etwas schwierig, da es hier keinen Scheduler mit festen Zeiten gibt, das hatte ich in einem anderen Post schon mal erfragt.

Wäre es eine Idee über MQTT ein Signal über den aktuellen Status des Ai-on-the-edge zu sende? Also z.b. "idle" wenn er nix macht und "processing" wenn er anfängt ein Foto zu machen. Daraus könnte man dann eine Automation bauen, die während der Zeit das Licht einschaltet.


2.) Mit einem Stück Plastikdiffusor aus einer LED Profilleiste die Flash LED bedecken

**Comment by Ingo-Knito** [comment]
Zu 1: Da der Strombedarf der LED gering ist: Warum dann nicht dauernd leuchten lassen? Der Strombedarf der smarten Steckdose dürfte höher sein als der für die LED.

Zu 2: Da habe ich viel probiert ohne es wirklich zu verbessern. Diffusoren in hell, dunkel, groß, klein, gebogen, usw. Alles nix. Ein kleiner Spiegel, der die ESP-Halterung anleuchtet noch eine Idee die ich nicht probiert habe. Aber vielleicht hat jemand anderes noch eine Idee?

**Comment by AdmiralRaccoon** [comment]
Gut zu wissen mit dem Diffusor, ich denke ich versuche das mal mit dem Dauerlicht.

**Reply by caco3** [reply]
Dauerlicht ist schon möglich, die LED wird über die Zeit minimal dunkler werden. Insb. wenn Du auf Max. Helligkeit stellts, könnte sie auch ausbrennen!

Hast Du als Diffusor einfach mal etwas Papier oder Karton (mit etwas Abstand) vor die LED gemacht? Also so, dass das Licht der LED nur noch indirekt auf die Zahlen fällt?

**Reply by AdmiralRaccoon** [reply]
ich hab heute morgen auch mal diverse Dinge probiert als Diffusor und komme auch zu dem gleichen Ergebnnis wie Ingo-Knito. Ich habs erstmal mit einem alten, weissen Strip gelöst, den ich an der 1. möglichen Stelle geschnitten habe (3 Leds). Sieht nicht schön aus und würde auch lieber ein zeitgesteuertes Licht nehmen, wenn die Aufnahme stattfindet, aber kommt vielleicht noch.
So bekomme ich zumindest ein Bild ohne Refklektionien in den Zahlen.


![image](https://user-images.githubusercontent.com/41727850/203991357-a8b0cd8a-2e91-4e9c-a799-7cb3900bd960.png)
![image](https://user-images.githubusercontent.com/41727850/203991791-a7ca6bf8-7d48-4463-ac11-5c98f808a36b.png)

**Comment by Ingo-Knito** [comment]
Das Problem an einem Diffusor ist, dass ein heller Spot immer übrig bleibt, da die interne LED eben leuchtet. Das sieht man immer in einer Reflexion, selbst wenn man die Helligkeit herunter regelt. Zumindest ich habe keinen guten Kompromiss aus Helligkeit und Lesbarkeit gefunden - und ich habe einiges probiert. Irgendwann wollte ich einen Kegel drucken, aber der Aufwand stand bei mir in keinem Verhältnis zu den ext. LED.

**Comment by AdmiralRaccoon** [comment]
Moin, um das nochmal kurz aufzugreifen...Hat das hier vielleicht auch schon einer probiert?

![image](https://user-images.githubusercontent.com/41727850/204223969-ff067cb2-a9ed-4aea-adc5-3c2d3524cb8a.png)

**Comment by Ingo-Knito** [comment]
Sieht optisch auf dem Bild eigentlich gut aus. Es löst aber nicht das Problem, dass das Licht der LED immer noch durchscheint - sonst macht eine Folie für ein Tablet wenig Sinn.
Eine Spiegelfolie - die das Licht komplett reflektiert - würde wahrscheinlich funktionieren. Allerdings muss man die auch irgendwie als Kegel im Gehäuse befestigen, sonst sieht die Kamera nichts mehr. Dies würde eine weitere Konstruktion nach sich ziehen.

Hmmm. Maximaler Aufwand vs. eine LED anlöten und in das Gehäuse kleben.

**Comment by AdmiralRaccoon** [comment]
So ganz lässt mich die Sache mit dem anlöten der LEDs auch nicht in Ruhe, aber dazu vorab gefragt: 
Ich habe diese Modelle der ESP32-CAM, könnte ich da überhaupt noch was externes anlöten, da das ja quasi komplett in dem Board steckt.
Hast du vielleicht mal ein Bild von deinem Setup?

![image](https://user-images.githubusercontent.com/41727850/204485318-80bd572f-9d90-44be-8ee6-8bc8d0aa505f.png)


Ich frage deswegen, weil ich mit der Wasseruhr noch nicht zufrieden bin, ich hab zwar jetzt keine Reflektionen in den Zahlen, wohl aber auf einem der analogen Zähler. Da kommt dann die Erkennung durcheinander, weil die rote Pfeilspitze durch die Reflektion nicht mehr zu erkennen ist.

![image](https://user-images.githubusercontent.com/41727850/204487001-1d79044d-1f32-4559-abf4-e17d6c070ac3.png)

**Reply by caco3** [reply]
Das aufgesteckte PCB (ESP32-CAM-MB) ist eigentlich nur eine Weiterleitung der Pins + USB-UART Converter. Die Pins für die LEDs kannst Du trotzdem noch oben (ESP32-CAM) oder unten (ESP32-CAM-MB) anlöten.

Zur Reflexion:
Würde es helfen, wenn Du den AUfbau 180° drehst? Ev. aber ja schon alles versucht :)

**Reply by AdmiralRaccoon** [reply]
Ja ich bin in der Tat schon einmal komplett rumgegangen ... :-D
Aber vielleicht mache ich es nochmal mit mehr Ruhe und kleinern Schritten

**Reply by Ingo-Knito** [reply]
> So ganz lässt mich die Sache mit dem anlöten der LEDs auch nicht in Ruhe, aber dazu vorab gefragt: Ich habe diese Modelle der ESP32-CAM, könnte ich da überhaupt noch was externes anlöten, da das ja quasi komplett in dem Board steckt. Hast du vielleicht mal ein Bild von deinem Setup?

![20221129_154059](https://user-images.githubusercontent.com/39132657/204581821-3b06c7b1-d6cc-481d-bfb2-4a068f781104.jpg)

Hier ein Bild vom vom ESP am PV-Zähler an dem man die drei Leitungen erkennt. Das Board schleift über den Stecker die PINs durch. Nicht die hübschesten Lötungen. Eigentlich kann ich das besser, aber funktioniert ;-)
Ich kann Dir aber gerne helfen und dir die Leitungen anlöten. Schreib mir einfach eine PM.

**Reply by AdmiralRaccoon** [reply]
Das ist super nett....komm ich gerne drauf zurück. Ich hab grade meinen Drucker mal angeschmissen um diese angewinkelten Halter zu drucken. Ich werde mal berichten ob das bei mir dann auch klappt. Dauert etwas.

**Reply by Ingo-Knito** [reply]
Die Winkelhalter finde ich eine spannende Lösung. Bei mir würden sie aber nicht zum Einsatz kommen, da ich noch Analogzeiger beim Wasserzähler habe. Das will ich mir gar nicht in der Erkennung vorstellen wegen dem Parallaxenfehler. Das ist schon rechtwinklig ein Problem - unabhängig von der Lichtquelle.
Für ein LC-Display aber sicher super.

**Reply by AdmiralRaccoon** [reply]
ja stimmt, da sagste wieder was...auf dem Thingiverse link hat der ja nur digitale zahlen und ich noch die analogen....ich teste einfach mal, bin mal gespannt was rauskommt

**Reply by Chris8er** [reply]
Ich habe auch analoge Anzeigen auf dem Wasserzähler. Funktioniert 1A mit dem thingiverse Adapter.

**Comment by Chris8er** [comment]
Ich konnte mir mit einem "Winkeladapter" helfen... Richtet man die Platine schräg auf den Zähler, kann man die Reflexionen vermeiden.

![zaehler](https://user-images.githubusercontent.com/58936905/204575767-f0e201b1-3225-4396-a99e-0924306eb18e.jpg)
![adapter](https://user-images.githubusercontent.com/58936905/204576191-16943634-b02f-410b-be36-424b7608210c.png)

Auf Thingiverse gibt es sowas auch für Wasserzähler:

https://www.thingiverse.com/thing:5198099

Schätze das ist die einfachste Lösung...

**Reply by AdmiralRaccoon** [reply]
ja cool, das sieht vielversprechend aus, danke.
Hättest du evtl. auch den link für den Gaszähler-Winkel, den hab ich jetzt auf Anhieb nicht finden können
